pkg/querier/queryrange/roundtrip.go

```diff
@@ -42,36 +42,46 @@ func NewTripperware(
 	schema chunk.SchemaConfig,
 	registerer prometheus.Registerer,
 ) (queryrangebase.Tripperware, Stopper, error) {
+	metrics := NewMetrics(registerer)
 
-	instrumentMetrics := queryrangebase.NewInstrumentMiddlewareMetrics(registerer)
-	retryMetrics := queryrangebase.NewRetryMiddlewareMetrics(registerer)
-	shardingMetrics := logql.NewShardingMetrics(registerer)
-	splitByMetrics := NewSplitByMetrics(registerer)
+	var (
+		c   cache.Cache
+		err error
+	)
+	if cfg.CacheResults {
+		c, err = cache.New(cfg.CacheConfig, registerer, log)
+		if err != nil {
+			return nil, nil, err
+		}
+		if cfg.Compression == "snappy" {
```

Comment:
Do we need to check other compression formats?

Reply:
This is copied from the other cache, I don't think we support anything else.